app-backend/db/src/main/scala/com/azavea/rf/database/UploadDao.scala

```diff
@@ -71,5 +74,24 @@ object UploadDao extends Dao[Upload] {
           project_id = ${upload.projectId},
           source = ${upload.source}
      """ ++ Fragments.whereAndOpt(Some(idFilter))).update.run
+    (for {
+      oldUpload <- oldUploadIO
+      newStatus <- upload.uploadStatus.pure[ConnectionIO]
+      nAffected <- recordUpdateIO
+      userPlatform <- UserDao.unsafeGetUserPlatform(oldUpload.owner)
+      owner <- UserDao.unsafeGetUserById(oldUpload.owner)
+     } yield (oldUpload, newStatus, nAffected, userPlatform, owner)) flatMap {
+      case (oldUpload: Upload, newStatus: UploadStatus, nAffected: Int, platform: Platform, owner: User) => {
+        (oldUpload.uploadStatus, newStatus, platform.publicSettings.emailIngestNotification, owner.emailNotifications) match {
+          case (UploadStatus.Processing, UploadStatus.Failed, true, true) =>
```

Comment:
I would flip these two cases.

Reply:
I flipped in the branches rather than in the case statements :+1: